src/containers/MintGenerative/StepCheckFiles.tsx

```diff
@@ -26,58 +27,80 @@ export const StepCheckFiles: StepComponent = ({ onNext, state }) => {
   const [check2, setCheck2] = useState<boolean>(false)
   const artworkIframeRef = useRef<ArtworkIframeRef>(null)
   const [features, setFeatures] = useState<RawTokenFeatures | null>(null)
+  const [params, setParams] = useState<any | null>([])
+  const [data, setData] = useState<Record<string, any>>({})
+
+  const inputBytes = useMemo<string | null>(() => {
+    const serialized = serializeParams(data, params)
+    if (serialized.length === 0) return null
+    return serialized
+  }, [data, params])
 
   const url = useMemo<string>(() => {
-    return ipfsUrlWithHash(state.cidUrlParams!, hash)
-  }, [hash])
+    return ipfsUrlWithHashAndParams(state.cidUrlParams!, hash, inputBytes)
+  }, [hash, inputBytes])
 
   const nextStep = () => {
     onNext({
       previewHash: hash,
+      previewInputBytes: inputBytes,
     })
   }
 
-  // attach event to window to get messages from
   useEffect(() => {
     const listener = (e: any) => {
-      if (e.data && e.data.id === "fxhash_getFeatures") {
-        if (e.data.data) {
-          setFeatures(e.data.data)
-        } else {
-          setFeatures(null)
+      if (e.data) {
+        if (e.data.id === "fxhash_getFeatures") {
+          if (e.data.data) {
+            setFeatures(e.data.data)
+          } else {
+            setFeatures(null)
+          }
+        }
+        if (e.data.id === "fxhash_getParams") {
+          if (e.data.data) {
+            setParams(e.data.data)
+          } else {
+            setParams(null)
+          }
```

Comment:
opinionated suggestion - easier to read with less nesting. this whole block could be

```
if (!e.data) return
if (e.data.id === "fxhash_getFeatures") setFeatures(e.data.data || null)
if (e.data.id === "fxhash_getParams") setParams(e.data.data || null)
```

Reply:
good catch 👍